libraries/AP_Math/definitions.h

```diff
@@ -36,8 +36,10 @@
 // GPS Specific double precision conversions
 // The precision here does matter when using the wsg* functions for converting
 // between LLH and ECEF coordinates.
+#ifdef ALLOW_DOUBLE_MATH_FUNCTIONS
```

Comment:
This should be #if and not #ifdef, because on gcc >= 7, if no previous definition was given, it will define  ALLOW_DOUBLE_MATH_FUNCTIONS = 0 and you will pass all #ifdef when we ideally don't want

Reply:
yep my bad it is " is not defined, evaluates to 0 [-Wundef]" ... so nothing wrong